Completed rename simAstrom->jointcal

parejkoj opened the pull request:
Changed filenames and module structure to eliminate /meas, as this is a task.

Renamed all files from simAstrom to jointcal

Replaced all simAstrom occurances with jointcal.

Included removal of meas:: namespace from .h, .cc, .py

timj commented:
When do we get python tests?

parejkoj commented:
That's much further down the line: DM-5297 is getting it to run under CI, at which point we can get some tests written. This is one of the (many) things that needs to be done to get to that point.

parejkoj commented:
Made one more commit: should have class names starting with uppercase J.

timj commented on python/lsst/jointcal/jointcal.py:
```diff
@@ -34,17 +34,17 @@
+from .jointcalLib import test, test2, jointcal, jointcalControl
 
-__all__ = ["SimAstromConfig", "SimAstromTask"]
+__all__ = ["jointcalConfig", "jointcalTask"]
```
Upper case?

parejkoj replied:
fixed and pushed.

timj commented on python/lsst/jointcal/jointcal.py:
```diff
@@ -55,11 +55,11 @@ class SimAstromConfig(pexConfig.Config):
+class JointcalTask(pipeBase.CmdLineTask):
  
-    ConfigClass = SimAstromConfig
+    ConfigClass = jointcalConfig
```
Isn't the class `JointcalConfig`?

parejkoj replied:
fixed and pushed.

timj commented:
It looks like there is still a `simAstrom3.py` file that should be `jointcal3.py`? Does this code run?

timj commented on bin/simAstrom.py:
```diff
@@ -21,6 +21,6 @@
 #
 import lsst.pex.logging as pexLog
-from lsst.meas.simastrom.simAstrom3 import SimAstromTask
+from lsst.jointcal.jointcal3 import jointcalTask
```
`jointcal3` is not mentioned anywhere else.

parejkoj replied:
I didn't touch these since I don't think they're relevant for anything. Maybe just delete them?

timj commented on bin/simAstromCoadd.py:
```diff
@@ -21,6 +21,6 @@
 #
 import lsst.pex.logging as pexLog
-from lsst.meas.simastrom.simAstromCoadd import SimAstromCoaddTask
+from lsst.jointcal.jointcalCoadd import jointcalCoaddTask
```
Case here?

parejkoj commented:
For  the record, there are simAstrom2 and simAstrom3, which I assume were other versions of the code while @boutigny and @PierreAstier were working on it. I've not renamed those, to make it more clear that they can be deleted in the near future, once I get everything working.

timj commented:
Probably need to move the `bin` scripts to a `shebang` directory. If you don't actually need the `bin` scripts then delete them.

parejkoj commented:
I'll delete them: these are all tasks to be run via retargeting, and we'll be getting proper tests eventually, and the bin/ stuff doesn't really do much.

parejkoj commented:
I'm asking for clarification about the (1)/2/3 numbered files. I'll delete/rename once I understand that better.

timj commented:
Ok. My confusion was that you had changed some places from `simAstrom` to `jointcal` but not the related files. Clearly doesn't matter if none of them are used but it would be nice to know which bits of the package are known to work and which is completely untested.

boutigny commented:
@parejkoj simAstrom3 --> jointcal3 is the one to be kept : running jointcal on every skymap tract and writing updated wcs
The others are leftover of previous development steps

timj commented:
So can we rename `simAstrom3` to straight `jointcal` or is the 3 meant to stay around?

parejkoj commented:
Tim: I've pushed the last set of changes for this (I hope!). I was able to run it on some data and get output (correct or not I don't yet know, but it's a start), so at least everything seems to be linking and importing correctly.

Do you want me to do some squashing of commits before you review, or would you rather look at it as-is?

boutigny commented:
Yes simAstrom3 can be renamed to jointcal.

On 03/03/2016 20:30, Tim Jenness wrote:
>
> So can we rename |simAstrom3| to straight |jointcal| or is the 3 meant 
> to stay around?
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/lsst/jointcal/pull/1#issuecomment-191926422>.
>

-- 
Dominique Boutigny

Laboratoire d'Annecy-le-Vieux de Physique des Particules
Université Savoie Mont Blanc - CNRS/IN2P3
Tel : 04.50.09.16.80 - 07.83.59.06.65
Bureau: 4-110

parejkoj commented:
Ok, squashing completed. Please take a look.

timj commented:
`ups/jointcal.cfg` needs to agree with the table file that `micro_cholmod` is optional.

timj commented:
The scripts in `bin/` are executable but do not have a shebang. If a shebang is going to be added then the files need to be moved to `bin.src` directory. Otherwise disable the executable bit.

parejkoj commented:
I'm punting on the details of the dependency list until RFC-153 is dealt with.

timj commented on bin/jointcalCoadd.py:
```diff
@@ -0,0 +1,5 @@
+# -*- python -*-
```
where did the license and copyright go?

parejkoj commented:
I thought the new policy for executable python was to put the `# -*- python -*-` a the top and let the installer handle adding the shebang? I will move them to bin.src though.

timj commented:
I'm a bit concerned as to the copyright status of this code. Two python files have old copyright years in them and are owned by LSST. There are some `CopyrightHeader` files in the root that look like they might be describing the license and copyright for all the code in this package but it's not entirely obvious. The C++ code is entirely lacking in copyright information and the README doesn't help. My first assumption was that the code was GPL with IN2P3 copyright. @boutigny ? This needs to be clarified before we merge this.

timj commented:
The installer does not do anything with shebangs. The shebang target in a Scons file in the `bin.src` directory rewrites `#!/usr/bin/env python` as appropriate as it copies the files to the `bin` directory.

timj commented on doc/simastrom.tex:
```diff
@@ -30,7 +30,7 @@
 
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
-\title{Documentation of the meas/simastrom package}
+\title{Documentation of the jointcal package}
```
Isn't this document describing the WCS part of jointcal? We should be explicit if we are assuming that `jointcal` is going to do photometry as well. Also, is that why this file was not renamed to `jointcal.tex`?

parejkoj replied:
Yes, this document is just about the WCS part of the algorithm. I'd rather not mess with the document, since from what I've heard it does not actually line up with the way the code is currently implemented (that's a separate issue).

timj replied:
I was thinking of a minor change of title to "Documentation of the astrometry component of the jointcal package".

parejkoj replied:
Fair point. Done.

timj commented:
I am still seeing "simastrom" in a couple of places: one in `doc/simastrom.tex` and the other in one of the iPython notebooks.

parejkoj commented:
The ipython notebooks don't actually touch the rest of the code, and appear to have been used for early algorithm testing. I'm not touching them.

I've renamed simastrom.tex and changed the one reference in it.

parejkoj commented:
I've fixed the shebang and licenses at the top of the python executables and move them to bin.src
